snoop/data/management/commands/filestats.py

```diff
@@ -0,0 +1,128 @@
+import sys
+from django.core.management.base import BaseCommand
+from ...logs import logging_for_management_command
+from ... import models
+from ... import collections
+from django.db.models import Count
+from django.db.models import Sum
+import re
+import mimetypes
+
+SUPPORTED_FILETYPES = {
+    'application/x-7z-compressed',
+    'application/zip',
+    'application/x-zip',
+    'application/x-rar',
+    'application/rar',
+    'application/x-gzip',
+    'application/gzip',
+    'application/x-bzip2',
+    'application/x-tar',
+    'application/x-hoover-pst', 
+    'application/mbox',
+    'application/pdf',
+
+    'text/plain',
+    'text/html',
+    'text/rtf',
+
+    'application/pdf',
+
+    'application/msword',
+    'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    'application/vnd.openxmlformats-officedocument.wordprocessingml.template',
+    'application/vnd.ms-word.document.macroEnabled.12',
+    'application/vnd.ms-word.template.macroEnabled.12',
+    'application/vnd.oasis.opendocument.text',
+    'application/vnd.oasis.opendocument.text-template',
+    'application/rtf',
+
+    'application/vnd.ms-excel',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.template',
+    'application/vnd.ms-excel.sheet.macroEnabled.12',
+    'application/vnd.ms-excel.template.macroEnabled.12',
+    'application/vnd.ms-excel.addin.macroEnabled.12',
+    'application/vnd.ms-excel.sheet.binary.macroEnabled.12',
+    'application/vnd.oasis.opendocument.spreadsheet-template',
+    'application/vnd.oasis.opendocument.spreadsheet',
+
+    'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+    'application/vnd.openxmlformats-officedocument.presentationml.template',
+    'application/vnd.openxmlformats-officedocument.presentationml.slideshow',
+    'application/vnd.ms-powerpoint',
+    'application/vnd.ms-powerpoint.addin.macroEnabled.12',
+    'application/vnd.ms-powerpoint.presentation.macroEnabled.12',
+    'application/vnd.ms-powerpoint.template.macroEnabled.12',
+    'application/vnd.ms-powerpoint.slideshow.macroEnabled.12',
+    'application/vnd.oasis.opendocument.presentation',
+    'application/vnd.oasis.opendocument.presentation-template',
+
+    'application/xhtml+xml',
+    'application/xml',
+    'text/xml',
+    'image/tiff',
+    'image/jpg',
+    'image/webp',
+    'image/heic',
+
+    'application/vnd.ms-outlook',
+    'application/vnd.ms-office',
+    'application/CDFV2',
+    'message/rfc822',
+    'message/x-emlx',
+}
+
+def truncateSize(size):
+    exponent = len(str(size))-1
+    divisor = 10**exponent
+    return int((size/divisor))*divisor
+
+class Command(BaseCommand):
+    help = "Display filetype stats."
+
+    def add_arguments(self, parser):
+        parser.add_argument('collection', type=str)
```

Comment:
instead of running this for only one collection, let's combine all collections on one server. I wouldn't wanna recombine these on my own from copy/pasted texts, right?

if you want you can set up an optional argument that you can specify multiple times called `collection` - when you have that argument, you only run it on the listed collections, not on all of them. `./manage.py filestats --collection testdata --collection testdata2`

it's fine if you collect top 100 results for each collection, then recombine the results (using dicts and sets like below) and then show a combined list that's actually longer than 100